src/main/java/com/splunk/ibm/mq/metricscollector/JobSubmitterContext.java

```diff
@@ -28,24 +28,24 @@
 public final class JobSubmitterContext {
 
   private final MonitorContextConfiguration monitorContextConfig;
-  private final CountDownLatch countDownLatch;
   private final MetricsCollectorContext collectorContext;
+  private final List<MetricsPublisher> pendingJobs;
 
   public JobSubmitterContext(
       MonitorContextConfiguration monitorContextConfig,
-      CountDownLatch countDownLatch,
-      MetricsCollectorContext collectorContext) {
+      MetricsCollectorContext collectorContext,
+      List<MetricsPublisher> pendingJobs) {
     this.monitorContextConfig = monitorContextConfig;
-    this.countDownLatch = countDownLatch;
     this.collectorContext = collectorContext;
+    this.pendingJobs = pendingJobs;
   }
 
   String getMetricPrefix() {
     return monitorContextConfig.getMetricPrefix();
   }
 
   Future<?> submitPublishJob(String name, MetricsPublisher publisher) {
-    MetricsPublisherJob job = new MetricsPublisherJob(publisher, countDownLatch);
+    MetricsPublisherJob job = new MetricsPublisherJob(publisher, null);
```

Comment:
I think overloading the constructor to fabricate an unrelated/disconnected internal latch is better than flinging nulls around.